modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1313,27 +1321,36 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 
 					if (setter_function == nullptr) {
 						push_error(vformat(R"(Setter "%s" not found.)", member.variable->setter_pointer->name), member.variable);
-
-					} else if (setter_function->parameters.size() != 1) {
-						push_error(vformat(R"(Function "%s" cannot be used as setter because of its signature.)", setter_function->identifier->name), member.variable);
-
-					} else if (!is_type_compatible(member.variable->datatype, setter_function->parameters[0]->datatype, true)) {
-						push_error(vformat(R"(Function with argument type "%s" cannot be used as setter for a property of type "%s".)", setter_function->parameters[0]->datatype.to_string(), member.variable->datatype.to_string()), member.variable);
-
 					} else {
-						has_valid_setter = true;
-
+						GDScriptParser::ParameterNode *value_arg = setter_function->parameters[setter_function->mandatory_parameter_count - 1];
+						if (setter_function->mandatory_parameter_count > 2) {
```

Comment:
```suggestion
						if (setter_function->mandatory_parameter_count == 0 || setter_function->mandatory_parameter_count > 2) {
```

Reply:
Fixed it.
Thanks for the help and patience!